s4e-web/src/app/views/map-view/state/utils/layers-loader.util.ts

```diff
@@ -0,0 +1,28 @@
+import { Observable, Observer } from 'rxjs';
+import { ImageWMS } from 'ol/source';
+
+export const IMAGE_WMS_LAYER = 'image-wms-layer-loader';
+export class ImageWmsLoader {
+  public start$: Promise<void>;
+  public end$: Promise<void>;
+
+  protected _error$: Promise<void>;
+  protected _isLoading = false;
+  protected _hasEnded = false;
+
+  constructor(source: ImageWMS) {
+    this._error$ = new Promise((resolve, reject) => {
+      source.on('imageloaderror', () => resolve());
```

Comment:
Are you 100% sure that subscribing via `.on` is alright? Won't it cause memory leak?

Reply:
`.on` in sync function, so there isn't any subscription occurring. If you mean passing the function to be called inside another one that will resolve it, I haven't seen any source that has talked about leaks in case of calling resolve in such a way.